docs/developer/plugins.md

```diff
@@ -0,0 +1,159 @@
+
+# Core plugins
+
+This page describes the build process for Nextflow core plugins. See the {ref}`plugins-page` page to learn how to create your own plugins outside of Nextflow.
+
+## Plugin subprojects
+
+Each plugin subproject has its own `build.gradle` which defines the main build actions and the required dependencies.
+
+The minimal dependencies are as follows:
+
+```groovy
+dependencies {
+    compileOnly project(':nextflow')
+    compileOnly 'org.slf4j:slf4j-api:1.7.10'
+    compileOnly 'org.pf4j:pf4j:3.4.1'
+
+    testImplementation project(':nextflow')
+    testImplementation "org.codehaus.groovy:groovy:3.0.5"
+    testImplementation "org.codehaus.groovy:groovy-nio:3.0.5"
+}
+```
+
+The plugin subproject directory name must begin with the prefix `nf-` and must include a file named `src/resources/META-INF/MANIFEST.MF` which contains the plugin metadata. The manifest content looks like the following:
+
+```
+Manifest-Version: 1.0
+Plugin-Class: the.plugin.ClassName
+Plugin-Id: the-plugin-id
+Plugin-Provider: Your Name or Organization
+Plugin-Version: 0.0.0
+```
+
+## Environment variables
```

Comment:
I went ahead and added `NXF_PLUGINS_DIR` and `NXF_PLUGINS_DEFAULT`, but not the other two because they are only used for Nextflow development and would be TMI for a Nextflow user